balancer-js/src/modules/pools/pool-types/concerns/weighted/exit.concern.integration.spec.ts

```diff
@@ -0,0 +1,321 @@
+
+const testExitPool = (poolId: string) => {
+  describe('exit execution', async () => {
+    // Setup chain
+    before(async function () {
+      const initialBalance = '100000';
+
+      this.timeout(20000);
+
+      const sdkConfig = {
+        network: Network.MAINNET,
+        rpcUrl,
+      };
+      // Using a static repository to make test consistent over time
+      const poolsProvider = new PoolsProvider(
+        sdkConfig,
+        new StaticPoolRepository(pools_14717479 as Pool[])
+      );
+      balancer = new BalancerSDK(
+        sdkConfig,
+        undefined,
+        undefined,
+        undefined,
+        poolsProvider
+      );
+      const poolSetup = await setupPool(poolsProvider, poolId);
+      if (!poolSetup) throw Error('Error setting up pool.');
+      pool = poolSetup;
+      tokensOut = pool.tokens;
+      await forkSetup(
+        balancer,
+        signer,
+        [pool.address],
+        [BPT_SLOT],
+        [parseFixed(initialBalance, 18).toString()],
+        jsonRpcUrl as string,
+        14717479 // holds the same state as the static repository
+      );
+      signerAddress = await signer.getAddress();
+    });
+
+    context('exitExactBPTIn', async () => {
+      const bptIn = parseFixed('10', 18).toString();
+
+      context('encoded data', async () => {
+        before(async function () {
+          this.timeout(20000);
+          [bptBalanceBefore, ...tokensBalanceBefore] = await updateBalances(
+            pool,
+            signer,
+            signerAddress,
+            balancer
+          );
+
+          const { to, data, minAmountsOut, maxBPTIn } =
+            pool.buildExitExactBPTIn(signerAddress, bptIn, slippage);
+          const tx = { to, data };
+
+          bptMaxBalanceDecrease = BigNumber.from(maxBPTIn);
+          tokensMinBalanceIncrease = minAmountsOut.map((a) =>
+            BigNumber.from(a)
+          );
+          const transactionResponse = await signer.sendTransaction(tx);
+          transactionReceipt = await transactionResponse.wait();
+          [bptBalanceAfter, ...tokensBalanceAfter] = await updateBalances(
+            pool,
+            signer,
+            signerAddress,
+            balancer
+          );
+        });
+
+        it('should work', async () => {
+          expect(transactionReceipt.status).to.eql(1);
+        });
+
+        it('tokens balance should increase by at least minAmountsOut', async () => {
+          for (let i = 0; i < tokensBalanceAfter.length; i++) {
+            expect(
+              tokensBalanceAfter[i]
+                .sub(tokensBalanceBefore[i])
+                .gte(tokensMinBalanceIncrease[i])
+            ).to.be.true;
+          }
+        });
+
+        it('bpt balance should decrease by max bptMaxBalanceDecrease', async () => {
+          expect(
+            bptBalanceBefore.sub(bptBalanceAfter).eq(bptMaxBalanceDecrease)
+          ).to.be.true;
+        });
+      });
+
+      context('params', async () => {
```

Comment:
@johngrantuk - Looking again at these tests scenarios (encoded vs params), they feel too similar. Almost as if we're testing if the encoding of the attributes was successful, since that's the only item that changes. Should we remove the redundancy?

Reply:
I know what you mean. In this case especially if the encoded case passes then the attributes must be correct so there's no need to test separately. In terms of making sure the user knows they can use encoded or params then that will be covered by the documentation. So yeah lets remove the redundancy.